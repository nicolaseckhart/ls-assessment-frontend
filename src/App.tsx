import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import HomeScreen from './screens/HomeScreen';
import ListPlacesScreen from './screens/places/ListPlacesScreen';
import ShowPlaceScreen from './screens/places/ShowPlaceScreen';
import NotFoundScreen from './screens/NotFoundScreen';

const App: React.FC = () => (
  <Router>
    <div className="App">
      <Switch>
        <Route exact path="/" component={HomeScreen} />
        <Route exact path="/places" component={ListPlacesScreen} />
        <Route exact path="/places/:id" component={ShowPlaceScreen} />
        <Route exact path="/*" component={NotFoundScreen} />
      </Switch>
    </div>
  </Router>
);

export default App;
