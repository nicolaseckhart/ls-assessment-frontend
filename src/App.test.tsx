import * as React from 'react';
import App from './App';
import { shallow, ShallowWrapper } from 'enzyme';

describe('App', () => {
  it('renders the app correctly', () => {
    const wrapper: ShallowWrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
