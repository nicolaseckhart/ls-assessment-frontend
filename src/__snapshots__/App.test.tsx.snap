// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app correctly 1`] = `
<div
  className="App"
>
  <h1>
    Welcome to this Assessment
  </h1>
</div>
`;
